test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java

```diff
@@ -262,32 +266,36 @@ public MockTransportService(
 
     public MockTransportService(
         Settings settings,
-        Transport transport,
+        StubbableTransport transport,
         ThreadPool threadPool,
         TransportInterceptor interceptor,
         Function<BoundTransportAddress, DiscoveryNode> localNodeFactory,
         @Nullable ClusterSettings clusterSettings,
-        String nodeId
+        TaskManager taskManager
```

Comment:
This constructor does not seem to be used anywhere before this change. Is it still the case with this PR? If so, can we remove it as part of this PR?